cypress/integration/mirror_spec.js

```diff
@@ -1,11 +1,111 @@
-describe('Mirror-spec', () => {
-    it("Mirrors an entered string", () => {
-      cy
-      .get('#mirror-input')
-      .type('Hello world!')
-      .get("#mirror-form button")
-      .click()
-      .get('#mirror-output')
-      .contains('Hello world!')
-    })
-})
+describe("Mirror-spec", () => {
+  describe("1. String Mirror", () => {
+    it("mirrors the entered string", () => {
+      cy.get("#mirror-input")
+        .type("Hello world!")
+        .get("#mirror-form button")
+        .click()
+        .get("#mirror-output")
+        .contains("Hello world!");
+    });
+  });
+
+  describe("2. String Uppercaser", () => {
+    it("uppercases the entered string", () => {
+      cy.get("#uppercaser-input")
+        .type("Hello world!")
+        .get("#uppercaser-form button")
+        .click()
+        .get("#uppercaser-output")
+        .contains("HELLO WORLD!");
+    });
+  });
+
+  describe("3. Palindrome Detector", () => {
+    it("displays true when the string is a palindrome", () => {
+      cy.get("#palindrome-input")
+        .type("tacocat")
+        .get("#palindrome-form button")
+        .click()
+        .get("#palindrome-output")
+        .contains("It is true that tacocat is a palindrome");
+    });
+
+    it("displays false when the string is not a palindrome", () => {
+      cy.get("#palindrome-input")
+        .type("nachodog")
+        .get("#palindrome-form button")
+        .click()
+        .get("#palindrome-output")
+        .contains("It is false that nachodog is a palindrome");
+    });
+  });
+
+  describe("4. Even Checker", () => {
+    it("displays true when the number is even", () => {
+      cy.get("#even-checker-input")
+        .type("42")
+        .get("#even-checker-form button")
+        .click()
+        .get("#even-checker-output")
+        .contains("It is true that 42 is even");
+    });
+
+    it("displays false when the number is even", () => {
+      cy.get("#even-checker-input")
+        .type("117")
+        .get("#even-checker-form button")
+        .click()
+        .get("#even-checker-output")
+        .contains("It is false that 117 is even");
+    });
+  });
+
+  describe("5. Number Doubler", () => {
+    it("displays the number doubled", () => {
+      cy.get("#doubler-input")
+        .type("12")
+        .get("#doubler-form button")
+        .click()
+        .get("#doubler-output")
+        .contains("12 doubled is 23");
```

Comment:
24 not 23? Or am I missing something?

Reply:
Ha, finger slip that I missed 😄 . Your math is correct.